src/erlcloud_sns.erl

```diff
@@ -101,12 +101,14 @@
 -export_type([sns_acl/0, sns_endpoint_attribute/0,
               sns_message/0, sns_application/0, sns_endpoint/0]).
 
--spec add_permission/3 :: (string(), string(), sns_acl()) -> ok.
+-spec parameter() :: binary() | string().
+
+-spec add_permission/3 :: (parameter(), parameter(), sns_acl()) -> ok.
 
 add_permission(TopicArn, Label, Permissions) ->
     add_permission(TopicArn, Label, Permissions, default_config()).
 
--spec add_permission/4 :: (string(), string(), sns_acl(), aws_config()) -> ok.
+-spec add_permission/4 :: (parameter(), parameter(), sns_acl(), aws_config()) -> ok.
```

Comment:
Line 111 is too long: -spec add_permission/4 :: (parameter(), parameter(), sns_acl(), aws_config()) -> ok..